diff --git a/skyquake/framework/core/api_utils/sockets.js b/skyquake/framework/core/api_utils/sockets.js
index 5e0b25b..93c6aa3 100644
--- a/skyquake/framework/core/api_utils/sockets.js
+++ b/skyquake/framework/core/api_utils/sockets.js
@@ -86,7 +86,7 @@ Subscriptions.prototype.subscribe = function(req, callback) {
       // No protocol was passed with the url in the body. Assume req.protocol is protocol and construct URL
       protocol = req.protocol || 'https';
       // Converting relative URL to full path.
-      origin = protocol + '://' + req.headers.host
+      origin = protocol + '://127.0.0.1:' + utils.getPortForProtocol(protocol);
     }
     var a = url.resolve(origin, req.baseUrl);
     var b = url.resolve(a, URL);
