diff --git a/jenkins/RO/start_build b/jenkins/RO/start_build
index 77f751d..aad20fb 100755
--- a/jenkins/RO/start_build
+++ b/jenkins/RO/start_build
@@ -33,6 +33,7 @@ if [ "$1" = "--notest" ]; then
 fi

 OSM_git_checkout "$@"
+git apply /opt/devops-patches/hot_fix.patch

 INFO "installing RO packages and dependencies from current repo (--noclone)"
 ./scripts/install-openmano.sh --noclone --force -q
